component/file_cache/file_cache.go

```diff
@@ -266,9 +270,31 @@ func (c *FileCache) OnConfigChange() {
 	c.cacheTimeout = float64(conf.Timeout)
 	c.policyTrace = conf.EnablePolicyTrace
 	c.offloadIO = conf.OffloadIO
+	c.maxCacheSize = conf.MaxSizeMB
 	c.policy.UpdateConfig(c.GetPolicyConfig(conf))
 }
 
+func (c *FileCache) StatFs() (*syscall.Statfs_t, bool, error) {
+	// cache_size = f_blocks * f_frsize/1024
+	// cache_size - used = f_frsize * f_bavail/1024
+	// cache_size - used = vfs.f_bfree * vfs.f_frsize / 1024
+	// if cache size is set to 0 then we have the root mount usage
+	cacheSize := c.maxCacheSize * MB
+	if cacheSize == 0 {
+		return nil, false, nil
+	}
+	usage := getUsage(c.tmpPath) * MB
+	available := cacheSize - usage
+	statfs := &syscall.Statfs_t{}
+
+	statfs.Frsize = 4096
```

Comment:
this part is generally 4k but depends on file system so on certain system we may end up showing wrong stats. Better to get cache paths stats from system and get this value from those returned values. we are doing "/" statfs in libfuse wrapper we can do the same here in golang for cache directory (mount point of cache path) and then take this value from there.

Reply:
done